src/sources/socket/udp.rs

```diff
@@ -92,7 +92,7 @@ pub fn udp(
             .expect("Failed to bind to udp listener socket");
 
         if let Some(receive_buffer_bytes) = receive_buffer_bytes {
-            if let Err(error) = udp::set_receive_buffer_size(&socket, receive_buffer_bytes) {
+            if let Err(error) = udp::set_receive_buffer_size(&socket, receive_buffer_bytes + 1) {
```

Comment:
I think this could be interpreted as a little misleading since the option is documented as:

> Configures the receive buffer size using the SO_RCVBUF option on the socket.

https://vector.dev/docs/reference/configuration/sources/vector/#receive_buffer_bytes

I don't think we need to modify this though? This configuration is for the entire receive buffer. I think we only need to add 1 when creating the buffer to receive into?